maps4fs/generator/texture.py

```diff
@@ -242,18 +245,33 @@ def layers(self, layers: list[Layer]) -> None:
     # pylint: disable=no-member
     def draw(self) -> None:
         """Iterates over layers and fills them with polygons from OSM data."""
-        for layer in self.layers:
+        layers = sorted(self.layers, key=lambda _layer: _layer.priority)
+        cumulative_image = None
+        for layer in layers:
             if not layer.tags:
                 self.logger.debug("Layer %s has no tags, there's nothing to draw.", layer.name)
                 continue
+            if layer.name == "grass":
+                continue
             layer_path = layer.path(self._weights_dir)
             self.logger.debug("Drawing layer %s.", layer_path)
-
             img = cv2.imread(layer_path, cv2.IMREAD_UNCHANGED)
+            cumulative_img = cumulative_image if cumulative_image is not None else img
+            img_mask = cv2.bitwise_not(cumulative_img) # converts black part of map to white
             for polygon in self.polygons(layer.tags, layer.width):  # type: ignore
                 cv2.fillPoly(img, [polygon], color=255)  # type: ignore
-            cv2.imwrite(layer_path, img)
+            output_img = cv2.bitwise_and(img, img_mask) # Wherever it's white on both images, it can draw
```

Comment:
Add more debug logs just to check shape, dtype, and so on.
And entries what we're doing, e.g. found base layer, will be...

Reply:
added logs for finding the base layer, tbh I don't even know what shape, dtype, and etc. area to properly do that.